src/app/app.module.ts

```diff
@@ -7,10 +7,11 @@ import { AppRoutingModule } from './app-routing.module'
 import { AppComponent } from './app.component'
 import { AdvertisementModule } from './modules/core/advertisement/advertisement.module'
 import { MaterialModule } from './modules/shared/material.module'
+import { DialogComponent } from './shared/components/dialog/dialog.component'
 import { PageNotFoundComponent } from './shared/components/pagenotfound/pagenotfound.component'
 
 @NgModule({
-  declarations: [AppComponent, PageNotFoundComponent],
+  declarations: [AppComponent, PageNotFoundComponent, DialogComponent],
```

Comment:
The `DialogComponent` imported in the app module looks quite a bad idea to me. It should go in a shared module most probably.